### DISCUSSION TITLE
ESP32 bootet nicht

### DISCUSSION THREAD

**Discussion by RicoUwe** [jomjol/AI-on-the-edge-device#1423] [Q&A]
Nach dem ich die Firmware auf den ESP32  mit dem WEB-Interface geladen habe (ohne Fehlermeldung) und danach den ESP neu starte bekomme ich folgende Fehlermeldungen:
E (883) psram: This chip is ESP32-D0WD
E (883) psram: PSRAM ID read error: 0xffffffff
E (883) cpu_start: Failed to init external RAM! 

Kann mir jemand einen Tip geben, was ich da falsch mache?

**Comment by Vaclar01** [comment]
Same Problem here. Also a SD-Karte(16GB Verbatim) with Wlan Settings is inserted, but no connection possible.

Installed with Webtool.
Hardware is: 
https://www.amazon.de/gp/product/B08P1NMPLL



######
Rebooting...
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:7664
load:0x40078000,len:15360
load:0x40080400,len:3744
entry 0x4008067c
I (27) boot: ESP-IDF 4.3.2 2nd stage bootloader
I (27) boot: compile time 20:02:47
I (27) boot: chip revision: 3
I (30) boot_comm: chip revision: 3, min. bootloader chip revision: 0
I (37) boot.esp32: SPI Speed      : 40MHz
I (42) boot.esp32: SPI Mode       : DIO
I (46) boot.esp32: SPI Flash Size : 4MB
I (51) boot: Enabling RNG early entropy source...
I (56) boot: Partition Table:
I (60) boot: ## Label            Usage          Type ST Offset   Length
I (67) boot:  0 nvs              WiFi data        01 02 00009000 00004000
I (74) boot:  1 otadata          OTA data         01 00 0000d000 00002000
I (82) boot:  2 phy_init         RF data          01 01 0000f000 00001000
I (89) boot:  3 ota_0            OTA app          00 10 00010000 001db000
I (97) boot:  4 ota_1            OTA app          00 11 001f0000 001db000
I (104) boot: End of partition table
I (109) boot_comm: chip revision: 3, min. application chip revision: 0
I (116) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=4df30h (319280) map
I (247) esp_image: segment 1: paddr=0005df58 vaddr=3ffb0000 size=020c0h (  8384) load
I (250) esp_image: segment 2: paddr=00060020 vaddr=400d0020 size=15e00ch (1433612) map
I (801) esp_image: segment 3: paddr=001be034 vaddr=3ffb20c0 size=029f8h ( 10744) load
I (806) esp_image: segment 4: paddr=001c0a34 vaddr=40080000 size=1c1f4h (115188) load
I (856) esp_image: segment 5: paddr=001dcc30 vaddr=50000000 size=00010h (    16) load
I (871) boot: Loaded app from partition at offset 0x10000
I (871) boot: Disabling RNG early entropy source...
I (883) psram: This chip is ESP32-D0WD
E (883) psram: PSRAM ID read error: 0xffffffff
E (883) cpu_start: Failed to init external RAM!
Re-enable cpu cache.

abort() was called at PC 0x40082dcf on core 0

Backtrace:0x40082c65:0x3ffe3ba0 0x4008f70d:0x3ffe3bc0 0x40096991:0x3ffe3be0 0x40082dcf:0x3ffe3c50 0x4007956d:0x3ffe3c80 |<-CORRUPTED


ELF file SHA256: 28356ed42df4c694

Rebooting...
#####

**Comment by caco3** [comment]
> E (883) psram: PSRAM ID read error: 0xffffffff
   E (883) cpu_start: Failed to init external RAM!

That indicates (for both of you) that your hardware is bad or not supported!
There is nothing we can do on this.

**Comment by sbesnard** [comment]
Hi all,

This is a very frustrating issue. I have the same, with this hardware : [amazon page](https://www.amazon.fr/gp/product/B08P1NMPLL/ref=ppx_yo_dt_b_asin_title_o00_s00)

But I am sure I am not the last to experience this, and I want to contribute.
I think the firmware needs to be adjusted to make it compatible, and that the adjustment could be small.
Can you point me to a set of resources to play with the firmware source code and compiler ? I don't think the fw source is in this repo.
Thanks
Serge

**Reply by sbesnard** [reply]
I already found [this](https://aithinker.blog.csdn.net/article/details/108000974) to compile (thanks google translate !!!), I just need to know how the current firmwares were tweaked to work with the software...

**Reply by caco3** [reply]
here the build is documented: https://github.com/jomjol/AI-on-the-edge-device/blob/rolling/code/README.md

also check our wiki!

**Reply by caco3** [reply]
just keep in mind, if PSRAM is <4MB, its unlikely you can get it working

**Comment by Vaclar01** [comment]
I bought the same hardwarepackage (2x ESP Cams "Diymore)) again, (same as @sbesnard posted).
This time it works without any problems.

So first time both ESP Cams were somehow broken, but with my second try(third Cam) it is runnung fine.
Didnt checked the last CAM yet.

**Comment by evilmumi** [comment]
i bought them here https://www.aliexpress.com/item/1005003804757059.html and have the same problem. If someone would find a solution it would be fine to share it.

**Reply by Vaclar01** [reply]
I assume it is an hardware fault in the production line. Maybe possible by repairing the hardware itself? As I understand it the RAM is physical broken.
But im not an expert.

**Reply by evilmumi** [reply]
i dont think. maybe its possible to change some parameter

**Comment by sbesnard** [comment]
No luck so far :/
Changed many parameters for spin in menuconfig, tried other boards, downgraded the excpressif32 version...

**Reply by evilmumi** [reply]
i found one webpage (forgot URL) which says something about camera version  - not sure if there is a connection.
Did you test some other projects with camera needed on that non-working ESP32CAM?

**Reply by sbesnard** [reply]
Yes, I installed Tasmota without a problem. I don't think it expects any external ram...

**Comment by jomjol** [comment]
If you are runnning a different firmware without problem, this is not a good test for the device. Any firmware, that uses no PSRAM will work.

AI-on-the-edge REALLY NEEDS 4MB of PSRAM!

I don't thinkg, that this is a problem of configuration or parametrization within the compile settings.

**Reply by evilmumi** [reply]
How can i find out -for sure- if my chip have this PSRAM - i got my money back from ALIEXPRESS but now i need a new one :(
> If you are runnning a different firmware without problem, this is not a good test for the device. Any firmware, that uses no PSRAM will work.
> 
> AI-on-the-edge REALLY NEEDS 4MB of PSRAM!
> 
> I don't thinkg, that this is a problem of configuration or parametrization within the compile settings.

**Reply by caco3** [reply]
Well, the easiest would be to get the hardware from a trustworthy vendor. I agree that is not practial with those ESP32-CAM boards...

So just try another seller and maybe do not take the cheapest one (although I am not sure if that helps at aliexpress)

**Reply by evilmumi** [reply]
i ordered a new one @ amazon. if it this will not working I can send it back

**Reply by evilmumi** [reply]
Mine from AZ DELIVERY was not good - i think a capacitor is beeping - annoying. but thats another problem.

**Comment by sbesnard** [comment]
In the meantime, I ordered this [item there](https://www.amazon.fr/gp/product/B093GSCBWJ/ref=ppx_yo_dt_b_asin_title_o00_s00?ie=UTF8&psc=1) and it works !

**Reply by sbesnard** [reply]
@evilmumi
